sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -134,11 +134,26 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
   }
 
   /**
-   * Returns a list of columns for the given table in the current database.
+   * Returns a list of columns for the temp view matching the given name, or for the given table in
+   * the current database.
    */
   @throws[AnalysisException]("table does not exist")
   override def listColumns(tableName: String): Dataset[Column] = {
```

Comment:
We are having a [test case](https://github.com/apache/spark/blob/c0ae6bc6ea38909730fad36e653d3c7ab0a84b44/sql/core/src/test/scala/org/apache/spark/sql/internal/CatalogSuite.scala#L239-L242) for `listColumns` on temporary views. However, it does not check results. Maybe we can correct it in this PR.